relayer/lavasession/consumer_session_manager.go

```diff
@@ -75,11 +76,34 @@ func (csm *ConsumerSessionManager) atomicReadCurrentEpoch() (epoch uint64) {
 	return atomic.LoadUint64(&csm.currentEpoch)
 }
 
+// validating we still have providers, otherwise reset valid addresses list
+func (csm *ConsumerSessionManager) validatePairingListNotEmpty() uint64 {
+	csm.lock.RLock() // lock read to validate length
+	numberOfResets := csm.numberOfResets
+	if len(csm.validAddresses) == 0 {
+		csm.lock.RUnlock()                // unlock read
+		csm.lock.Lock()                   // lock write
+		if len(csm.validAddresses) == 0 { // re verify it didn't change while waiting for lock.
+			utils.LavaFormatWarning("Provider pairing list is empty, resetting state.", nil, nil)
+			csm.validAddresses = make([]string, csm.pairingAddressesLength)
+			copy(csm.validAddresses, csm.pairingAddresses) // reset the list of valid addresses
```

Comment:
initialising should be done with the same code from update providers, not a copy of the same lines, create a small function that creates a new valid addresses from pairingAddresses and call it here and in updateAllProviders